x-pack/plugins/security_solution/public/common/components/event_details/osquery_tab.tsx

```diff
@@ -69,29 +66,29 @@ export const useOsqueryTab = ({
     skip: shouldEarlyReturn,
   });
 
-  const expandedEventFieldsObject = rawEventData
-    ? (expandDottedObject(rawEventData.fields) as ExpandedEventFieldsObject)
-    : undefined;
-
-  const responseActions =
-    expandedEventFieldsObject?.kibana?.alert?.rule?.parameters?.[0].response_actions;
+  if (shouldEarlyReturn) {
+    return;
+  }
 
-  const osqueryResponseActions = useMemo(
-    () =>
-      responseActions?.filter(
-        (responseAction) => responseAction.action_type_id === RESPONSE_ACTION_TYPES.OSQUERY
-      ),
-    [responseActions]
+  const osqueryResponseActions = responseActions.filter(
+    (responseAction) => responseAction.action_type_id === RESPONSE_ACTION_TYPES.OSQUERY
   );
 
-  if (!osqueryResponseActions?.length || shouldEarlyReturn) {
+  if (!osqueryResponseActions?.length) {
     return;
   }
 
   const actionItems = actionsData?.data.items || [];
 
   const ruleName = expandedEventFieldsObject?.kibana?.alert?.rule?.name;
 
+  const content = (
```

Comment:
Could we memoize this with `useMemo`?

Reply:
Thanks, good question! I was thinking about doing it, but we can't because it would break the hooks rule (earlyReturn before). I could move earlyReturn below these variables, but then it wouldn't be an early return in the end. What do you think?

Reply:
Ah, got it! I would say that, if the value is memoized and nothing changes, the early return could be right after it but I'm ok leaving it as it is